Style/Frontend/index.md

```diff
@@ -0,0 +1,1451 @@
+        ? (totalChange / purchaseValue) * 100
+        : 'N/A';
+    const portfolioValue = holdings
+      .filter((p) => p.totalValue)
+      .reduce((a, b) => a + b.totalValue, 0);
+    const portfolioPercentage =
+      totalValue !== 'N/A' ? (totalValue / portfolioValue) * 100 : 'N/A';
+
+    ...
+
+  }
+  ```
+
+## Testing
+
+- New components should always be accompanied by a corresponding jest test. Learn about [jest](https://jestjs.io/) and read through our tests to get an idea of how we utilize it
+- Bug fixes should always be accompanied by a corresponding regression test
+- Be cautious about stubs and mocks - they can make your tests more brittle
+- Strive to write many small pure functions, and minimize where mutations occur
+- 100% test coverage is a good goal to strive for, even if it’s not always practical to reach it
+
+## Comments & Docstrings
+
+- Make use of `TODO`s, `FIXME`s and `NOTE`s where it makes sense
+
+  ```jsx
+  // TODO for things that need to be done
+  // FIXME for things that don't work as they should
+  // NOTE for implementation details, edge cases, etc
+
+  const calculateEPS = () => {
+    // TODO: finish implementing eps calculation
+    return 0;
+  }
+
+  // FIXME: not returning expected values
+  const square(x) => x * 2;
+
+  // NOTE: this component expects xyz to be present in order to work properly
+  const ComplicatedComponent = ({props}) => { ... } 
+  ```
+
+- If you comment something out because you don't need it, delete it instead
+
+  ```jsx
+  // bad
+  const styles = StyleService.create({
+    featureContainer: {
+      flex: 1,
+      // justifyContent: 'space-between'
+    }
+  });
+
+  // an exception
+  // it is acceptable to comment code that will be uncommented in the future
+  // just make sure to note it with a FIXME or TODO
+  const styles = StyleService.create
+    // TODO: Style for a currently disabled component that needs some work
+    // featureContainer: {
+    //  flex: 1,
+    // }
+  });
+  ```
+
+- New components and screens should always have a good docstring with a general description, list of params and what they are, and any important info you want to communicate to other devs who will come across it
+
+  ```jsx
+  /**
+   * Widget component for showing data with a particular color and style
+   * @param data {Object} data object to be rendered, should be in xyz format otherwise it will not render properly
+   * @param [color='highlight'] {String} optional color string to color data with
+   * @param [style={}] {Style} optional style to apply to container
+   */
+  const Widget = ({data, color = 'highlight', style = {}}) => { ... }
+  ```
+
+## Imports & Exports
+
+- Use modules
+
+  ```jsx
+  // bad
+  const MyLib = require('@/path/MyLib');
+  module.exports = MyLib.doThing;
+  
+  // getting there
+  import MyLib from '@/path/MyLib';
+  export default MyLib.doThing;
+  
+  // best
+  import {MyLib} from '@/path/MyLib';
```

Comment:
This should be `import {doThing} ...`

Seems like the ones below also need to change